pkg/kinflate/tree/node.go

```diff
@@ -20,39 +20,66 @@ import (
 	"k8s.io/kubectl/pkg/kinflate/types"
 )
 
-type namePrefixType string
+type NamePrefixType string
 
-type objectLabelsType map[string]string
+type ObjectLabelsType map[string]string
 
-type objectAnnotationsType map[string]string
+type ObjectAnnotationsType map[string]string
 
-type resourcesType types.KObject
+type ResourcesType types.KObject
 
-type patchesType types.KObject
+type PatchesType types.KObject
 
-type configmapsType types.KObject
+type ConfigmapsType types.KObject
 
-type secretsType types.KObject
+type SecretsType types.KObject
 
-// ManifestNode groups (possibly empty) manifest data with a (possibly empty)
-// set of manifest nodes.
-// data in one node may refer to data in other nodes.
+type PackagesType []*ManifestData
+
+// ManifestData contains all the objects loaded from the filesystem according to
+// the Manifest Object.
+// Data in one node may refer to data in other nodes.
 // The node is invalid if it requires data it cannot find.
 // A node is either a base app, or a patch to the base, or a patch to a patch to the base, etc.
-type ManifestNode struct {
-	data     *manifestData
-	children []*ManifestNode
+type ManifestData struct {
+	// Name of the manifest
+	Name string
+
+	NamePrefix        NamePrefixType
+	ObjectLabels      ObjectLabelsType
+	ObjectAnnotations ObjectAnnotationsType
+	Resources         ResourcesType
+	Patches           PatchesType
+	Configmaps        ConfigmapsType
+	Secrets           SecretsType
+
+	Packages PackagesType
 }
 
-// manifestData contains all the objects loaded from the filesystem according to
-// the Manifest Object.
-type manifestData struct {
-	name              string
-	namePrefix        namePrefixType
-	objectLabels      objectLabelsType
-	objectAnnotations objectAnnotationsType
-	resources         resourcesType
-	patches           patchesType
-	configmaps        configmapsType
-	secrets           secretsType
+func (md *ManifestData) AllResources() error {
+	err := types.Merge(md.Resources, md.Configmaps)
+	if err != nil {
+		return err
+	}
+	return types.Merge(md.Resources, md.Secrets)
+}
+
+func (md *ManifestData) Inflate() error {
+	for _, pkg := range md.Packages {
+		err := pkg.Inflate()
+		if err != nil {
+			return err
+		}
+		err = types.Merge(md.Resources, pkg.Resources)
```

Comment:
May be separate this merge from pkg-inflation ?

merge them at the end when all the sub-packages have been inflated ? (so we don't cause any side-effect in case of errors)